### PR TITLE
chore: add tests for updated `Account.fund` logic

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/2266

### CODE REVIEW

File: packages/fuel-gauge/src/funding-transaction.test.ts [hunk 1]
```diff
@@ -232,4 +236,109 @@ describe(__filename, () => {
...
+  it('should ensure a partially funded Transaction will require only missing funds', async () => {
+    const receiver = Wallet.generate({ provider });
+    const wallet1 = Wallet.generate({ provider });
+    const wallet2 = Wallet.generate({ provider });
+
+    const totalInBaseAsset = 15_000;
+    const totalInAssetA = 20_000;
+    const partiallyInAssetA = totalInAssetA / 2;
+
+    // wallet1 does not have enough funds to cover total required in assetA and
```
Comment (maschad) [inline packages/fuel-gauge/src/funding-transaction.test.ts:249]: Not understanding this comment

File: packages/fuel-gauge/src/funding-transaction.test.ts [hunk 2]
```diff
@@ -232,4 +236,109 @@ describe(__filename, () => {
...
+    ]);
+
+    const baseAssetResource = partiallyResources.find((r) => r.assetId === baseAssetId);
+    const assetAResource = partiallyResources.find((r) => r.assetId === assetA);
+
+    // expect to have the correct amount of resources, not enough to cover the TX needs
+    expect(baseAssetResource?.amount.toString()).toBe(totalInBaseAsset.toString());
+    expect(assetAResource?.amount.toString()).toBe(partiallyInAssetA.toString());
+
+    transactionRequest.addResources(partiallyResources);
```
Comment (maschad) [inline packages/fuel-gauge/src/funding-transaction.test.ts:282]: Maybe I am misinterpreting the issue but shouldn't this test essentially ensure that the `fund` method is only called  when resources if they are missing? Or perhaps the method should reject an attempt to add resources if they are present?

**Review by maschad — COMMENTED** [review]
It's difficult to evaluate what is being tested here

File: packages/fuel-gauge/src/funding-transaction.test.ts (hunk 1 above)
Comment (Torres-ssf) [reply packages/fuel-gauge/src/funding-transaction.test.ts]: @maschad is it more clear now? https://github.com/FuelLabs/fuels-ts/pull/2267/commits/bb25beffc6021b04462596c4d88ccad695b6c998

File: packages/fuel-gauge/src/funding-transaction.test.ts (hunk 2 above)
Comment (Torres-ssf) [reply packages/fuel-gauge/src/funding-transaction.test.ts:289]: @maschad No, the issue is to validate that the `Account.fund` method will not try to fetch and add more funds to a TX that is already funded.

**Review by maschad — APPROVED** [review]
Thanks for the clarification, nice work!

**Review by danielbate — APPROVED** [review]

**Pull request merged**